assets/js/components/DateRangeSelector.js

```diff
@@ -19,40 +19,92 @@
 /**
  * WordPress dependencies
  */
-import { useCallback } from '@wordpress/element';
+import { useCallback, useEffect, useRef, useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
-import { Option, Select } from '../material-components';
+import DateRangeIcon from '../../svg/date-range.svg';
+import Menu from './menu';
 import { getAvailableDateRanges } from '../util/date-range';
 import { STORE_NAME as CORE_USER } from '../googlesitekit/datastore/user/constants';
+import Button from './button';
+import { map } from 'lodash';
 const { useSelect, useDispatch } = Data;
 
 function DateRangeSelector() {
 	const ranges = Object.values( getAvailableDateRanges() );
 	const dateRange = useSelect( ( select ) => select( CORE_USER ).getDateRange() );
 	const { setDateRange } = useDispatch( CORE_USER );
-	const onChange = useCallback( ( index, item ) => {
-		setDateRange( item.dataset.value );
-	}, [ ranges ] );
+
+	const [ menuOpen, toggleMenu ] = useState( false );
```

Comment:
Let's call this `setMenuOpen` rather than `toggleMenu`. A function named `toggleMenu()` implies to me that it'll switch it from the current state to the inverse state, but that's not what this does 🙂

Reply:
Yes I like this terminology.